core/lib/protobuf_config/src/proto/api.proto

```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
```

Comment:
wdyt about using proto2 and having required fields?

Reply:
Please do not use required fields. It's always a disaster in long term

Reply:
We have restricted out code generator so that proto binary encoding is canonical (so that we can sign proto binary messages). For simplicity, only proto3 syntax is supported and all fields are required to be either optional or repeated. Canonical encoding is obviously not required for configs, however it would take some extra work to make code generator distinguish these use cases.

Reply:
and yes, having required fields in protos is considered an antipattern, because required fields can never be deprecated.